### PR TITLE
Add quick start guide in README.md

### PR DESCRIPTION
**Pull request by fmuyassarov**
This PR adds quick start guide to set up Metal3-dev-env and provision machines.

### CODE REVIEW

**Comment by russellb** [comment]
what about https://github.com/metal3-io/metal3-io.github.io/blob/master/_includes/try-it.md ?

File: README.md [hunk 1]
```diff
@@ -1,17 +1,262 @@
...
+
+[![asciicast](https://asciinema.org/a/295294.png)](https://asciinema.org/a/VanYI09TcGdod7YnII1cBqFhY?speed=3&theme=tango)
+
+Please follow the steps below to build Metal3 dev environment.
+```bash
+# clone the Metal³-dev-env github repo
+$ git clone https://github.com/metal3-io/metal3-dev-env.git
+$ cd ./metal3-dev-env
+# export environment variables depending on your need
+$ export CAPI_VERSION=v1alpha1
```
Comment (maelk) [inline README.md]: there's a problem with the versions. here you have v1a1

File: README.md [hunk 2]
```diff
@@ -1,17 +1,262 @@
...
+You should see the following four pods are created and running in ``metal3``
+namespace
+
+```bash
+$ kubectl get pods -n metal3
+NAME                                         READY   STATUS    RESTARTS   AGE
+cabpk-controller-manager-5c67dd56c4-jdrgk    2/2     Running   0          25m
+capbm-controller-manager-7f9b8f96b7-mb4cm    2/2     Running   0          25m
+capi-controller-manager-798c76675f-sg5fs     1/1     Running   0          25m
+metal3-baremetal-operator-5cbbd7b87d-f4ktc   6/6     Running   0          25m
```
Comment (maelk) [inline README.md]: that's v1a2

File: README.md [hunk 3]
```diff
@@ -1,17 +1,262 @@
...
+------------
+
+Once you have successfully built Metal³-dev-env, you can start creating a
+cluster and set of machines, where one is Kubernetes master node and the second
+is a worker node. Following script will apply
+[Cluster custom resource](https://github.com/metal3-io/metal3-dev-env/blob/master/crs/v1alpha2/cluster.yaml)
+(CR) to provision a cluster in ``metal3`` namespace.
+
+```bash
+$ ./scripts/v1alpha1/create_cluster.sh
```
Comment (maelk) [inline README.md]: that's nopt v1a1, that's v1a2 in the path

File: README.md [hunk 4]
```diff
@@ -1,17 +1,262 @@
...
+$ kubectl get cluster -n metal3
+NAME    PHASE
+test1   provisioned
+```
+
+Next, run ``create_controlplane.sh`` script to create a master machine,
+baremetalmachine and its associated KubeadmConfig in  ``metal3`` namespace.
+
+```bash
+$ ./scripts/v1alpha1/create_controlplane.sh
```
Comment (maelk) [inline README.md]: same here, v1a2

File: README.md [hunk 5]
```diff
@@ -1,17 +1,262 @@
...
+NAME     STATUS   PROVISIONING STATUS   CONSUMER               BMC                         HARDWARE PROFILE   ONLINE   ERROR
+node-0   OK       ready                                        ipmi://192.168.111.1:6230   unknown            true     
+node-1   OK       provisioning          test1-controlplane-0   ipmi://192.168.111.1:6231   unknown            true     
+```
+
+Next, run the ``create_worker.sh`` script to create the worker machine,
+baremetalmachine and its associated KubeadmConfig in ``metal3`` namespace.
+
+```bash
+$ ./scripts/v1alpha1/create_worker.sh
```
Comment (maelk) [inline README.md]: and here too, v1a2

File: README.md [hunk 6]
```diff
@@ -1,17 +1,262 @@
...
+```
+
+Deprovision Cluster and Machines
+------------
+
+You can run the following scripts to deprovision your target cluster and
+machines.
+
+```bash
+$ ./scripts/v1alpha1/delete_worker.sh
```
Comment (maelk) [inline README.md]: here too

File: README.md [hunk 7]
```diff
@@ -1,17 +1,262 @@
...
+machinedeployment.cluster.x-k8s.io "test1-md-0" deleted
+baremetalmachinetemplate.infrastructure.cluster.x-k8s.io "test1-md-0" deleted
+kubeadmconfigtemplate.bootstrap.cluster.x-k8s.io "test1-md-0" deleted
+
+$ ./scripts/v1alpha1/delete_controlplane.sh
```
Comment (maelk) [inline README.md]: here too

File: README.md [hunk 8]
```diff
@@ -1,17 +1,262 @@
...
+machine.cluster.x-k8s.io "test1-controlplane-0" deleted
+baremetalmachine.infrastructure.cluster.x-k8s.io "test1-controlplane-0" deleted
+kubeadmconfig.bootstrap.cluster.x-k8s.io "test1-controlplane-0" deleted
+
+$ ./scripts/v1alpha1/delete_cluster.sh
```
Comment (maelk) [inline README.md]: here too

**Comment by fmuyassarov** [comment]
> what about https://github.com/metal3-io/metal3-io.github.io/blob/master/_includes/try-it.md ?
should we update try-it.md as well accordingly or keep only try-it.md and keep changes of this PR for the future book's quick start page?  I added this here for now hoping that it will help for new folks to set up dev-env easily. 
On another note, should we then store upcoming docs that will be added to the book under /doc for now?

**Comment by hardys** [comment]
> should we update try-it.md as well accordingly or keep only try-it.md and keep changes of this PR for the future book's quick start page?

Is there a reason for not simply updating the existing try-it.md with any missing content from this PR?

IME having multiple quick-start guides results in confusion for new users and it's more work to maintain, so we should aim to have exactly one guide IMO (I personally don't have a particularly strong opinion on which repo it lives in though)

**Comment by fmuyassarov** [comment]
> > should we update try-it.md as well accordingly or keep only try-it.md and keep changes of this PR for the future book's quick start page?
> 
> Is there a reason for not simply updating the existing try-it.md with any missing content from this PR?
> 
> IME having multiple quick-start guides results in confusion for new users and it's more work to maintain, so we should aim to have exactly one guide IMO (I personally don't have a particularly strong opinion on which repo it lives in though)

I agree with you @hardys .  I'll simply update try-it.md then and close this PR. 
Thanks

**Pull request closed without merging**